walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/constraints/GroupConstraint.kt

```diff
@@ -41,19 +41,18 @@ class GroupConstraint(
             try {
                 constraint.matches(verifiedId)
             } catch (exception: WalletLibraryException) {
-                exception.message?.let { validationExceptions.add(it) }
+                validationExceptions.add("$exception")
             }
         }
 
         when (constraintOperator) {
             GroupConstraintOperator.ANY -> {
                 if (constraints.size == validationExceptions.size)
-                    throw NoMatchForAnyConstraintsException("None of the constraints match.")
+                    throw NoMatchForAnyConstraintsException("None of the constraints match $validationExceptions.")
```

Comment:
nit: maybe add a comma before validationExceptions? Or make NoMatchForAnyConstraintsException take in the list of exceptions like:  NoMatchForAnyConstraintsException(message: String, exceptions: [Exceptions])

Reply:
Is the assumption that when thrown, app developers will either select the group (Any) or iterate through the group to find the non-fulfilled one (All) for UI treatment? In which case the error message is only for debugging right? We don't need to add the validationExceptions for the app developer to use directly?

Reply:
Added exception list so that developer can use it for UI.